src/Core/src/Graphics/CompareExtensions.cs

```diff
@@ -1,10 +1,12 @@
 ﻿using System;
+using System.Runtime.CompilerServices;
 
 namespace Microsoft.Maui.Graphics
 {
 	internal static class CompareExtensions
 	{
-		public static T Clamp<T>(this T value, T min, T max) where T : IComparable<T>
+		[MethodImpl(MethodImplOptions.AggressiveInlining)]
+		public static DateTime Clamp(this DateTime value, DateTime min, DateTime max)
```

Comment:
So one thing to note is that `[MethodImpl(MethodImplOptions.AggressiveInlining)]` increases app size as the AOT images will be larger (`libaot-Microsoft.Maui.Graphics.dll.so`).

I would maybe not add it here if you didn't see it specifically improve something.